### ISSUE TITLE
How does the pretrain dataset consist of?

### ISSUE THREAD

**Issue by lucasjinreal** [OpenGVLab/InternVL#49]
How does the pretrain dataset consist of?

Especially wonder how does the OCR part data prepared.

**Comment by czczup** [comment]
Hi, the pre-training dataset contains about 450M samples, including Laion-EN, Laion-COCO, COYO, CC12M, CC3M, SBU, GRIT, Objects365, OpenImages, Wukong, and so on. This is a subset filtered by the quality of the caption. But I usually don't do a full epoch of training on this dataset, because I find training only 40~50M samples can get good results.

OCR data is generated using open-source OCR recognition models to create pseudo-labels. The images used are the same as those above.

**Comment by lucasjinreal** [comment]
Any plan to opensource the ocr pretrain data?

**Comment by czczup** [comment]
Thanks for your attention. Currently, there are no plans to release OCR data. However, you can easily create a similar dataset, for example, by using [PaddleOCR](https://github.com/PaddlePaddle/PaddleOCR?tab=readme-ov-file). 

I think there are two main reasons why OCR performance is not as good as QwenVL's. 

First, the domain of OCR data is too narrow; we mainly use pseudo labels on images from natural scenes, and there is too little document data; while Qwen uses a large amount of document and web page data. 

Second, a resolution of 448 is not sufficient for recognizing some smaller text, and it is necessary to increase to a larger resolution to enhance OCR capabilities.

**Comment by lucasjinreal** [comment]
@czczup thanks for the suggestions, however, there are some traditional OCR opensource data out there, how to utilizing them to construct your simillar data? Note that I think pretrain image-text pairs might not the same way compare in sft.

**Comment by czczup** [comment]
Traditional OCR datasets can be transformed into instruction-following datasets. For example, in the traditional OCR dataset, a data sample is an image with OCR ground truths.

We can construct a lot of questions, e.g., "please identify all the words in the image."
Then, convert the OCR ground truths to the answer, e.g., "The text in the image includes:\ntext1\ntext2\ntext3"

**Comment by mumtozee** [comment]
> Traditional OCR datasets can be transformed into instruction-following datasets. For example, in the traditional OCR dataset, a data sample is an image with OCR ground truths.
> 
> We can construct a lot of questions, e.g., "please identify all the words in the image." Then, convert the OCR ground truths to the answer, e.g., "The text in the image includes:\ntext1\ntext2\ntext3"

Thank you for you answer! I couldn't understand, whether you mix OCR samples in an instruction-following format with image captioning samples during pre-training. From the code of InternVL-Chat-1.5 I couldn't figure out if the pre-training dataset is also in the instruction-following format.

**Comment by xmy0916** [comment]
> Hi, the pre-training dataset contains about 450M samples, including Laion-EN, Laion-COCO, COYO, CC12M, CC3M, SBU, GRIT, Objects365, OpenImages, Wukong, and so on. This is a subset filtered by the quality of the caption. But I usually don't do a full epoch of training on this dataset, because I find training only 40~50M samples can get good results.
> 
> OCR data is generated using open-source OCR recognition models to create pseudo-labels. The images used are the same as those above.

where can i get this pretrain dataset?

**Comment by nemonameless** [comment]
Any plan to opensource the ocr pretrain data now ? @czczup thanks

**Comment by vanpersie32** [comment]
@czczup  Do you have any plan to open source the pertaining datasets? Thanks

**Comment by vanpersie32** [comment]
> Hi, the pre-training dataset contains about 450M samples, including Laion-EN, Laion-COCO, COYO, CC12M, CC3M, SBU, GRIT, Objects365, OpenImages, Wukong, and so on. This is a subset filtered by the quality of the caption. But I usually don't do a full epoch of training on this dataset, because I find training only 40~50M samples can get good results.
> 
> OCR data is generated using open-source OCR recognition models to create pseudo-labels. The images used are the same as those above.

Thanks for your great work. Can you tell us how to preprocess the data including data filtering, data deduplication and data selection?
